dora/core/server/worker/src/test/java/alluxio/worker/dora/PagedFileReaderTest.java

```diff
@@ -0,0 +1,196 @@
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertTrue;
+
+import alluxio.AlluxioURI;
+import alluxio.Constants;
+import alluxio.PositionReaderTest;
+import alluxio.client.file.cache.CacheManager;
+import alluxio.client.file.cache.CacheManagerOptions;
+import alluxio.client.file.cache.DefaultPageMetaStore;
+import alluxio.client.file.cache.PageMetaStore;
+import alluxio.client.file.cache.PageStore;
+import alluxio.client.file.cache.evictor.CacheEvictor;
+import alluxio.client.file.cache.evictor.CacheEvictorOptions;
+import alluxio.client.file.cache.evictor.FIFOCacheEvictor;
+import alluxio.client.file.cache.store.MemoryPageStore;
+import alluxio.client.file.cache.store.MemoryPageStoreDir;
+import alluxio.client.file.cache.store.PageStoreDir;
+import alluxio.client.file.cache.store.PageStoreOptions;
+import alluxio.client.file.cache.store.PageStoreType;
+import alluxio.conf.Configuration;
+import alluxio.conf.InstancedConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.network.protocol.databuffer.CompositeDataBuffer;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.util.io.BufferUtils;
+
+import com.google.common.collect.ImmutableList;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.embedded.EmbeddedChannel;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Random;
+
+/**
+ * Unit test of {@link PagedFileReader}.
+ */
+@RunWith(Parameterized.class)
+public class PagedFileReaderTest {
+
+  @Parameterized.Parameter
+  public int mFileLen;
+  @Rule
+  public TemporaryFolder mTemporaryFolder = new TemporaryFolder();
+  private final InstancedConfiguration mConf = Configuration.copyGlobal();
+  private String mTestFileName;
+  private PagedFileReader mPagedFileReader;
+  private UnderFileSystem mLocalUfs;
+  private byte[] mTestData;
+  private final int mMinTestNum = 50;
+  private final Random mRandom = new Random();
+  private final EmbeddedChannel mEmbeddedChannel = new EmbeddedChannel();
+
+  private PositionReaderTest mPositionReaderTest;
+
+  @Parameterized.Parameters(name = "{index}-{0}")
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][] {
+        {0},
+        {1},
+        {128},
+        {256},
+        {666},
+        {5314},
+        {Constants.KB - 1},
+        {Constants.KB},
+        {Constants.KB + 1},
+        {64 * Constants.KB - 1},
+        {64 * Constants.KB},
+        {64 * Constants.KB + 1},
```

Comment:
add cases where file size is large than the size of a page, i.e. 1 MB by default.